src/test/kotlin/org/opensearch/replication/bwc/BackwardsCompatibilityIT.kt

```diff
@@ -0,0 +1,94 @@
+package org.opensearch.replication.bwc;
+
+import org.assertj.core.api.Assertions
+import org.junit.Assert
+import org.opensearch.action.DocWriteResponse
+import org.opensearch.action.admin.cluster.health.ClusterHealthRequest
+import org.opensearch.action.get.GetRequest
+import org.opensearch.action.index.IndexRequest
+import org.opensearch.client.RequestOptions
+import org.opensearch.client.indices.CreateIndexRequest
+import org.opensearch.replication.MultiClusterAnnotations
+import org.opensearch.replication.MultiClusterRestTestCase
+import org.opensearch.replication.StartReplicationRequest
+import org.opensearch.replication.startReplication
+import org.opensearch.replication.stopReplication
+import org.opensearch.test.OpenSearchTestCase.assertBusy
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+import java.util.stream.Collectors
+
+
+const val LEADER = "bwcLeader"
+const val FOLLOWER = "bwcFollower"
+
+@MultiClusterAnnotations.ClusterConfigurations(
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${LEADER}0"),
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${FOLLOWER}0"),
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${LEADER}1"),
+    MultiClusterAnnotations.ClusterConfiguration(clusterName = "${FOLLOWER}1")
+)
+class BackwardsCompatibilityIT : MultiClusterRestTestCase() {
```

Comment:
Can we explore having CCR on while upgrade happen? That way we get better coverage over serialization etc and mimics most user scenarios.

Reply:
Perhaps cleanup @After can be skipped in BWC tests and only clean them in @AfterClass?

Reply:
Finally let us include some javadoc on how this test is supposed to be written and also call out how this test gets invoked (and how it gets skipped in regular gradle task etc).